### PR TITLE
[FIX] Report: handle PermissionError when trying to save

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
https://sentry.io/biolab/orange3/issues/251030241/

##### Description of changes
Now shows a windows with error message and logs error as well.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/251030241/

File: Orange/canvas/report/owreport.py [hunk 1]
```diff
@@ -337,14 +340,14 @@ def open_report(self):
 
         try:
             report = self.load(filename)
-        except (IOError, AttributeError) as e:
+        except (IOError, AttributeError):
```
Comment (janezd) [inline Orange/canvas/report/owreport.py]: I tried to open a file with extension .report, but which was not actually a report file. It crashes with `UnpicklingError`. Perhaps you should add it to list, or even catch the general `Exception` here.

File: Orange/canvas/report/owreport.py [hunk 2]
```diff
@@ -337,14 +340,14 @@ def open_report(self):
...
             message_critical(
-                 self.tr("Could not load an Orange Report file"),
-                 title=self.tr("Error"),
-                 informative_text=self.tr("An unexpected error occurred "
-                                          "while loading '%s'.") % filename,
-                 exc_info=True,
-                 parent=self)
+                self.tr("Could not load an Orange Report file"),
+                title=self.tr("Error"),
+                informative_text=self.tr("An unexpected error occurred "
```
Comment (janezd) [inline Orange/canvas/report/owreport.py]: Remove *unexpected* (all errors are unexpected) and probably also the article, hence "Error occurred while loading '%s'.

File: Orange/canvas/report/owreport.py [hunk 3]
```diff
@@ -337,14 +340,14 @@ def open_report(self):
...
+                                         "while loading '%s'.") % filename,
```
Comment (janezd) [inline Orange/canvas/report/owreport.py]: Use formatting with `{}` and `format`, not the C-style `%`. I would say the the old-style is permitted in Orange only for consistency in the modules that (still) use the old style.

File: Orange/canvas/report/owreport.py [hunk 4]
```diff
@@ -162,13 +162,13 @@ def _setup_ui_(self):
 
         class PyBridge(QObject):
             @pyqtSlot(str)
-            def _select_item(myself, item_id):
+            def _select_item(self, item_id):
```
Comment (jerneju) [inline Orange/canvas/report/owreport.py]: Shame on me. I blindly followed lint's complaints.

**Pull request merged**